REFERENCE.md

```diff
@@ -0,0 +1,464 @@
+
+##### `ssl_version`
+
+Data type: `Nrpe::Sslversion`
+
+The SSL Version to use.  The default of `TLSv1.2+` is the most secure option available at time of writing.  Avoid having to set it to a lower value if possible.
+
+Default value: $nrpe::params::ssl_version
+
+##### `ssl_ciphers`
+
+Data type: `Array[String[1]]`
+
+An array of ciphers that should be allowed by NRPE.  The defaults are for RSA keys and were taken from https://github.com/ssllabs/research/wiki/SSL-and-TLS-Deployment-Best-Practices.
+
+Default value: $nrpe::params::ssl_ciphers
+
+##### `ssl_client_certs`
+
+Data type: `Enum['no','ask','require']`
+
+This options determines client certificate usage.
+
+Default value: $nrpe::params::ssl_client_certs
+
+##### `ssl_log_startup_params`
+
+Data type: `Boolean`
+
+Whether to log startup SSL/TLS parameters.
+
+Default value: `false`
+
+##### `ssl_log_remote_ip`
+
+Data type: `Boolean`
+
+Whether to log remote IP address of SSL client.
+
+Default value: `false`
+
+##### `ssl_log_protocol_version`
+
+Data type: `Boolean`
+
+Whether to log SSL/TLS version of connections.
+
+Default value: `false`
+
+##### `ssl_log_cipher`
+
+Data type: `Boolean`
+
+Whether to log which encryption cipher is being used for SSL connections.
+
+Default value: `false`
+
+##### `ssl_log_client_cert`
+
+Data type: `Boolean`
+
+Whether to log if an SSL client has presented a certificate.
+
+Default value: `false`
+
+##### `ssl_log_client_cert_details`
+
+Data type: `Boolean`
+
+Whether to log details of client SSL certificates.
+
+Default value: `false`
+
+##### `config`
+
+Data type: `Stdlib::Absolutepath`
+
+**Private** You should not need to override this parameter.
+
+Default value: $nrpe::params::nrpe_config
+
+##### `include_dir`
+
+Data type: `Stdlib::Absolutepath`
+
+**Private** You should not need to override this parameter.
+
+Default value: $nrpe::params::nrpe_include_dir
+
+##### `provider`
+
+Data type: `Optional[String[1]]`
+
+**Private** You should not need to override this parameter.
+
+Default value: $nrpe::params::nrpe_provider
+
+##### `service_name`
+
+Data type: `String[1]`
+
+**Private** You should not need to override this parameter.
+
+Default value: $nrpe::params::nrpe_service
+
+## Defined types
+
+### nrpe::command
+
+Installs NRPE commands
+
+#### Examples
+
+##### Install a command called `check_users`
+
+```puppet
+nrpe::command { 'check_users':
+  ensure  => present,
+  command => 'check_users -w 5 -c 10',
+}
+```
+
+#### Parameters
+
+The following parameters are available in the `nrpe::command` defined type.
+
+##### `name`
+
+The name of the command.
+
+##### `command`
+
+Data type: `String[1]`
+
+The command plugin to run and its arguments.
+
+##### `ensure`
+
+Data type: `Enum['present', 'absent']`
+
+Whether to install or remove the command.
+
+Default value: present
+
+##### `file_mode`
+
+Data type: `Optional[Stdlib::Filemode]`
+
+The mode to use for the command file.  Defaults to `$nrpe::command_file_default_mode`.
```

Comment:
Below it says it defaults to `undef`. Perhaps it should say it'll use that var when it's undef?

Reply:
Fixed.